driver/src/gnosis_safe_gas_station.rs

```diff
@@ -0,0 +1,106 @@
+//! Retrieves gas prices from the https://safe-relay.gnosis.io/ api call `/v1/gas-station`.
+
+use anyhow::Result;
+use ethcontract::U256;
+use isahc::http::uri::Uri;
+use isahc::prelude::*;
+use serde::Deserialize;
+use std::time::Duration;
+use uint::FromDecStrErr;
+
+/// The default uri at which the gas station api is available under.
+pub const DEFAULT_URI: &str = "https://safe-relay.gnosis.io/api/v1/gas-station/";
+
+/// Result of the api call. Prices are in wei.
+#[derive(Deserialize, Debug, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GasPrice {
+    pub last_update: String,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub lowest: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub safe_low: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub standard: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fast: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fastest: U256,
+}
+
+#[cfg_attr(test, mockall::automock)]
+pub trait GasPriceEstimating {
+    /// Retrieves gas prices from the Gnosis Safe Relay api.
+    fn estimate_gas_price(&self) -> Result<GasPrice>;
+}
+
+pub struct GasPriceEstimatingImpl {
+    client: HttpClient,
+    uri: Uri,
+}
+
+impl GasPriceEstimatingImpl {
+    pub fn new(request_timeout: Duration, api_uri: &str) -> Result<GasPriceEstimatingImpl> {
+        let client = HttpClient::builder().timeout(request_timeout).build()?;
+        let uri: Uri = api_uri.parse()?;
+        Ok(GasPriceEstimatingImpl { client, uri })
+    }
+}
+
+impl GasPriceEstimating for GasPriceEstimatingImpl {
+    fn estimate_gas_price(&self) -> Result<GasPrice> {
+        Ok(self.client.get(&self.uri)?.json()?)
+    }
+}
+
+fn deserialize_u256_from_string<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    U256::from_dec_str(&s)
+        .map_err(|err| format!("{}: {}", uint_error_to_string(err), s))
+        .map_err(serde::de::Error::custom)
+}
+
+fn uint_error_to_string(err: FromDecStrErr) -> &'static str {
+    match err {
+        FromDecStrErr::InvalidCharacter => "FromDecStrErr: invalid character",
+        FromDecStrErr::InvalidLength => "FromDecStrErr: invalid length",
+    }
+}
```

Comment:
Can we submit a PR to `uint` crate to implement the `Error` trait to avoid these shenanigans?

Reply:
I am going to do that.